### DISCUSSION TITLE
How can I activate logging to a file under Windows?

### DISCUSSION THREAD

**Discussion by Ekopalypse** [rust-lang/rust-analyzer#16381] [Q&A]
Starting rust-analyzer with `-v --log-file d:\rust-analyzer.txt --no-log-buffering` creates a log file, but nothing is written to this file. Repeating `-v -v -v` has no effect. 
What am I missing?
I am particularly interested in logging the lsp communication between rust-analyzer and my editor.
Thx.

**Comment by Veykril** [comment]
you might need to set the `RA_LOG` env var to something (e.g `INFO), though I think we have a default value set for that so that should already log something 🤔

**Reply by lnicola** [reply]
`RA_LOG=lsp_server=debug RA_LOG_FILE=x` should also work (you cat set those with `set RA_LOG=lsp_server=debug` in `cmd`, without extra spaces).

**Comment by Ekopalypse** [comment]
Thank you, yes, using the environment variables works.
